### PR TITLE
Add PR template and commit guidelines

### PR DESCRIPTION
**Pull request by venkata-lakshmi-penna**
Signed-off-by: Venkata Lakshmi Dhatri Penna <venkata.lakshmi.dhatrix.penna@intel.com>

### CODE REVIEW

**Review by soda480 — CHANGES_REQUESTED** [review]
I think it would be cleaner if you should add the labels checkbox under the PR checklist heading instead of its own heading.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## PR Checklist
+Please check if your PR fulfills the following requirements:
+
+- [ ] The commit message follows our guidelines: https://github.com/edgexfoundry/app-functions-sdk-go/blob/master/.github/CONTRIBUTING.md
```
Comment (lranjbar) [inline .github/PULL_REQUEST_TEMPLATE.md]: We need to update the link here to https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## PR Type
+What kind of change does this PR introduce?
+<!-- Please check the one that applies to this PR using "x". -->
+
+- [ ] Bugfix
+- [ ] Feature
+- [ ] Code style update (formatting, local variables)
+- [ ] Refactoring (no functional changes, no api changes)
+- [ ] Build related changes
```
Comment (lranjbar) [inline .github/PULL_REQUEST_TEMPLATE.md]: This check box is redundant... we can remove it.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+* **fix**: A bug fix
+* **perf**: A code change that improves performance
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, etc)
+* **test**: Adding missing tests or correcting existing tests
+
+### Scope
+The scope should be the name of the module or package affected (as perceived by the person reading the changelog generated from commit messages.
+
+The following is the list of suggested scopes:
```
Comment (lranjbar) [inline .github/CONTRIBUTING.md:58]: The scopes for ci-management would be different. We can remove all of the modules/packages scopes.

Suggested ci-management scopes:
* **jenkins-config**
* **jjb**
* **packer**
* **pipeline**

@ernestojeda @jamesrgregg @soda480 @MightyNerdEric Some feedback on this is appreciated.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Closes: #123
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **build**: Changes that affect the CI/CD pipline or build system or external dependencies (example scopes: travis, jenkins, makefile)
```
Comment (lranjbar) [inline .github/CONTRIBUTING.md]: Build is redundant probably just remove... everything in ci-management is related to this.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix
+* **perf**: A code change that improves performance
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, etc)
+* **test**: Adding missing tests or correcting existing tests
```
Comment (lranjbar) [inline .github/CONTRIBUTING.md]: Don't think we will have tests in ci-management?

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+Previously remote logging failed due to improper initialization of the logging client. This commit fixes the initialization to properly support remote logging.
+
+Closes: #123
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
```
Comment (lranjbar) [inline .github/CONTRIBUTING.md:42]: Add types -
* **script**: Shell scripts that are used in Jenkins builds.
* **jjb-template**: New JJB template or changes to an existing template
* **jjb-job**: New JJB job or changes to an existing JJB job

Feedback on this idea is appreciated. Unsure which of these we want in types and scopes

@ernestojeda @jamesrgregg @soda480 @MightyNerdEric

**Review by lranjbar — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (venkata-lakshmi-penna) [reply .github/PULL_REQUEST_TEMPLATE.md]: okay

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (jamesrgregg) [reply .github/CONTRIBUTING.md:42]: I'm fine with the suggestion

**Review by jamesrgregg — CHANGES_REQUESTED** [review]
I would approve with recommended changes @lranjbar & @soda480 have already mentioned / suggested.

**Comment by venkata-lakshmi-penna** [comment]
> I would approve with recommended changes @lranjbar & @soda480 have already mentioned / suggested.

>Update the recommended changes

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] Refactoring (no functional changes, no api changes)
+- [ ] Documentation content changes
+- [ ] Other... Please describe:
+
+## Issue Number:
+
+## Sandbox Testing
+Test Links :
+
+Test Instructions:
```
Comment (ernestojeda) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think we said no test instructions. Links to the sandbox job(s) should be sufficient.

**Review by ernestojeda — COMMENTED** [review]
See my comment about test instructions

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (venkata-lakshmi-penna) [reply .github/PULL_REQUEST_TEMPLATE.md]: okay

**Review by lranjbar — APPROVED** [review]

**Comment by lranjbar** [comment]
recheck

**Comment by lranjbar** [comment]
recheck

**Review by jamesrgregg — APPROVED** [review]

**Review by jamesrgregg — APPROVED** [review]

**Review by soda480 — APPROVED** [review]

File: shell/docker-push.sh [hunk 8]
```diff
@@ -2,5 +2,9 @@
 # Ensure we fail the job if any steps fail
 set -e -o pipefail
 
+
```
Comment (eb-oss) [inline shell/docker-push.sh:5]: The entire change to this file should be removed. What is the need for four extra blank lines here?

**Review by eb-oss — DISMISSED** [review]
Commit message needs to be in the imperative.

Also, this PR has four commits. It should probably be rebased down to one. A single extra merge commit is not the end of the world, but four commits for one change is excessive.

File: shell/docker-push.sh (hunk 8 above)
Comment (venkata-lakshmi-penna) [reply shell/docker-push.sh:5]: This change was made when the JJB Verify wasn't getting triggered. IT asked to create a dummy change to some file for the verify job  to get triggered.

**Comment by jamesrgregg** [comment]
> Commit message needs to be in the imperative.
> 
> Also, this PR has four commits. It should probably be rebased down to one. A single extra merge commit is not the end of the world, but four commits for one change is excessive.

I updated this just now.

**Pull request merged**